core/lib/core/studies/study_public.ex

```diff
@@ -52,22 +77,83 @@ defmodule Core.Studies.StudyPublic do
     struct(Core.Studies.StudyPublic, opts)
   end
 
-  def get_byline(%Study{} = study, %SurveyTool{} = survey_tool) do
-    date =
-      if SurveyTool.published?(survey_tool) do
-        timestamp = Timestamp.humanize(survey_tool.published_at)
-        "#{dgettext("eyra-survey", "published.true.label")}: #{timestamp}"
-      else
-        timestamp = Timestamp.humanize(survey_tool.inserted_at)
-        "#{dgettext("eyra-survey", "created.label")}: #{timestamp}"
-      end
+  def get_devices(survey_tool) do
+    [:desktop, :phone, :tablet]
+    |> Enum.filter(&survey_tool[String.to_atom("#{&1}_enabled")])
+  end
 
+  def get_byline(%Study{} = study, _survey_tool) do
     authors =
       study
       |> Studies.list_authors()
       |> Enum.map(& &1.fullname)
       |> Enum.join(", ")
 
-    date <> " - #{dgettext("eyra-survey", "by.author.label")} " <> authors
+    "#{dgettext("eyra-survey", "by.author.label")}: " <> authors
+  end
+
+  defp get_organisation_name(survey_tool) do
+    case get_organisation_id(survey_tool) do
+      nil ->
+        nil
+
+      id ->
+        organisation =
+          Marks.instances()
+          |> Enum.find(&(&1.id === String.to_existing_atom(id)))
+
+        organisation.label
+    end
+  end
+
+  defp get_organisation_id(survey_tool) do
```

Comment:
This could be written as two functions with pattern matches on the args:

`defp get_organisation_id(%{marks: [first_mark|_]}), do: first_mark`

`defp get_organisation_id(_), do: nil`